solutions/usaco-859.mdx

```diff
@@ -0,0 +1,76 @@
+---
+id: usaco-859
+source: USACO Silver 2018 December
+title: Convention II
+author: Ryan Chou
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_convention2_silver_dec18.html)
+
+Due to the low bounds on $N$, we can naively simulate the events.
+
+## Implementation
+**Time Complexity:** $\mathcal{O}(N \log N)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <cstdio>
+#include <algorithm>
+#include <vector>
+#include <set>
+using namespace std;
+
+int main() {
+	freopen("convention2.in", "r", stdin);
+	freopen("convention2.out", "w", stdout);
+
+	int n;
+	vector<vector<int>> cows;
+	cin >> n;
+
+	for (int i = 0; i < n; i++) {
+		int start, duration;
+		cin >> start >> duration;
+		cows.push_back({i, start, duration});
+	}
+
+	// sort by arrival time.
+	sort(cows.begin(), cows.end(), [](const vector<int> &a, const vector<int> &b) {
+		return a[1] < b[1];
+	});
+
+	int time = 0, cur = 0, ans = 0;
+
+	// sorted by priority.
+	set<vector<int>> waiting;
+
+	// as long as we haven't processed all cows or there are still cows waiting.
+	while (cur < n || waiting.size() > 0) {
+		
```

Comment:
is this empty line intentional?

Reply:
yeah, for the comment. should i remove it?

Reply:
nah i was just wondering

Reply:
![image](https://user-images.githubusercontent.com/81596991/141228812-ece958a5-2ecb-4dd5-b07a-ba3f32984b98.png)

Reply:
what

Reply:
![image](https://user-images.githubusercontent.com/81596991/141230283-7c526de5-bb47-4b0d-9949-a35d28fb1c4b.png)

Reply:
...you don't need that whitespace for that comment.